users/decorator.go

```diff
@@ -23,25 +23,40 @@ import (
 	"github.com/trackit/trackit2/routes"
 )
 
-type WithAuthenticatedUser struct{}
+type RequireAuthenticatedUser struct{}
 
-type withAuthenticatedUserArgumentKey uint
+type authenticatedUserArgumentKey uint
 
 const (
-	AuthenticatedUser = withAuthenticatedUserArgumentKey(iota)
+	AuthenticatedUser            = authenticatedUserArgumentKey(iota)
+	TagRequireUserAuthentication = "require:userauth"
 )
 
-func (d WithAuthenticatedUser) Decorate(h routes.IntermediateHandler) routes.IntermediateHandler {
+func (d RequireAuthenticatedUser) Decorate(h routes.Handler) routes.Handler {
+	h.Func = d.getFunc(h.Func)
+	h.Documentation = d.getDocumentation(h.Documentation)
+	return h
+}
+
+func (_ RequireAuthenticatedUser) getFunc(hf routes.HandlerFunc) routes.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request, a routes.Arguments) (int, interface{}) {
 		auth := r.Header["Authorization"]
 		tx := a[db.Transaction].(*sql.Tx)
 		if auth != nil && len(auth) == 1 {
 			tokenString := auth[0]
 			if user, err := testToken(tx, tokenString); err == nil {
```

Comment:
Here you check if the `err` is `nil` but you never handle it in the case that it is not `nil`